src/core/mac/wakeup_tx_scheduler.cpp

```diff
@@ -88,7 +88,7 @@ Mac::TxFrame *WakeupTxScheduler::PrepareWakeupFrame(Mac::TxFrames &aTxFrames)
 
     target.SetExtended(mWedAddress);
     source.SetExtended(Get<Mac::Mac>().GetExtAddress());
-    radioTxUs = static_cast<uint32_t>(Get<Radio>().GetNow()) + (mTxTimeUs - TimerMicro::GetNow());
+    radioTxUs = mTxTimeUs - TimerMicro::GetNow();
```

Comment:
I wonder if we should add a check that `mTxTimeUs` has not expired (is `>= GetNow()`) to ensure this subtraction is valid. We can do this in a future PR.

The `PrepareWakeupFrame()` method seems to be used by `Mac` when it starts `kOperationTransmitWakeup`. It is technically possible that something blocks the start and delays it until after `mTxTimeUs` has passed.

Reply:
We can add:

```cpp
   TimeMicro nowUs = TimerMicro::GetNow();

    ...

   VerifyOrExit(mTxTimeUs >= nowUs);
   radioTxUs = (mTxTimeUs - nowUs);
```

We should read `now` and keep it in a local variable to avoid possibly reading different values as the function executes.